### ISSUE TITLE
uasyncio raises "local variable referenced before assignment"

### ISSUE THREAD

**Issue by amirgon** [micropython/micropython#6566]
When trying to cancel a task that is waiting on `wait_for` with timeout, `wait_for` tries to access `ret` although it is not assigned.

Example:

```python
import uasyncio

async def task1():
    await uasyncio.sleep(3)

async def task2(t1):
    await uasyncio.sleep(1)
    t1.cancel()

async def task3(t1):
    try:
        await uasyncio.wait_for(t1, 2)
    except uasyncio.TimeoutError:
        pass

t1 = uasyncio.create_task(task1())
t2 = uasyncio.create_task(task2(t1))
t3 = uasyncio.create_task(task3(t1))
uasyncio.Loop.run_forever()
```

Result (unix port "dev" variant):
```
Task exception wasn't retrieved
future: <Task> coro= <generator object 'task3' at 7fae42272b00>
Traceback (most recent call last):
  File "uasyncio/core.py", line 1, in run_until_complete
  File "<stdin>", line 12, in task3
  File "uasyncio/funcs.py", line 1, in wait_for
NameError: local variable referenced before assignment
```

Probably related to this:

https://github.com/micropython/micropython/blob/05f95682e7ddfb08c317e83826df9a1d636676f3/extmod/uasyncio/funcs.py#L20

But in my case, it's not due to timeout.

**Comment by dpgeorge** [comment]
See related #5797

**Comment by amirgon** [comment]
Duplicate of #5797.
Closing.

**Comment by dpgeorge** [comment]
Fixed by b505971069333a373d9f0c12138b64dab83fed72
